### ISSUE TITLE
Kendo angular ui grid column filters: Edits of filters block for large data sets, instead of being asynchronously queuing the filter search.

### ISSUE THREAD

**Issue by HansGuckInDieLuft** [telerik/kendo-angular#1036]
## I'm submitting a...

* Regression report 
* Bug report 
* Suggestion for improvement
* Documentation issue or request

## Environment

Package versions:

```

```

Browser:

- Chrome (desktop) version XX
- Chrome (Android) version XX
- Chrome (iOS) version XX
- Firefox version XX
- Safari (desktop) version XX
- Safari (iOS) version XX
- IE version XX
- Edge version XX

System:
- TypeScript version: XX 
- Node version: XX  
- Platform:

**Comment by tsvetomir** [comment]
Hi, can you please update the issue template with more details and a runnable sample? Thanks!

**Comment by HansGuckInDieLuft** [comment]
Hello Tsvetomir,

I am sorry that I cannot give you a coded sample as that would mean to give you
access to our business data, even if it is only from our staging system.
The table however, has about 30000 rows, and the number of columns fetched is 11.
At an ODATA page size of 300 rows the blocking experience is huge: 
Just filter a text column by the word "the". When you begin typing into the filter at moderate speed, the "t" search will block the successing letters typed to appear in the
filter. What has to be done for the word "the" to be seached for from the beginning is to paste it in from the clipboard. When you set the page size to 30, the blocking experience is hardly noticeable.

All that behaviour is certainly testable from your side without the business data.
As to the Grid itself, it is data bound to the ODATA source by the custom binding directive from your web site, so plain vanilla.

Much the same line and perhaps the same cause: Having the grid defined wider as the 
page size, a horizontal scrollbar is shown. However scrolling (to the right) will work first, as soon as data are being shown.

Kind Regards
Detlef

**Comment by tsvetomir** [comment]
Understood, the specific data is not really important in this case.

My suggestion is to implement filter debouncing as demonstrated in #536. This should remove the bulk of the requests that are detrimental to the user experience.

**Comment by HansGuckInDieLuft** [comment]
Hello Tsvetomir,

Thank you very much: I'll try on Monday at work and will reply.

Kind Regards and have a good weekend
Detlef

**Comment by HansGuckInDieLuft** [comment]
Hello Tsvetomir,

no: that did not help: in that code the init function of the BindingDirective is overridden calling a pageChange-event.
In it, take() is expected to be part of the base class as a method, but that's missing in my base class:
What's happened? Is it, that the sample is relying on an older version?

Likewise: Debounce() cannot be found: Where does it come from? Why does the demo code not mention it?
Do I need to implement it myself?

In that thread, you proposed the right things: publish debounce-techniques as an extension to the standard demo.
And you got just the wrong answer by rkonstantinov: Leave it/Hide details!

As such, Kendo proved to me to be a time- and effort-killer, rather than a help.


Kind Regards
Detlef

Detlef Hüttenbach
Senior Developer
Information Services

Phone: +49 2273 597 8472 IP-Phone: 78472, Fax: +49 2273 597 222
Email: detlef.huettenbach@computacenter.com<mailto:detlef.huettenbach@computacenter.com>

Europaring 34-40, 50170 Kerpen, Nordrhein-Westfalen, Germany

**Comment by tsvetomir** [comment]
The sample had a few minor issues, but overall it should work fine. I've put together [a demo](http://plnkr.co/edit/Gb3t9JDRQWdqKzBlxsTG?p=preview) based on it to make sure everything works as it should.

The debounceTime operator is part of RxJS. The import was missing in the original snippet, but it still worked due to the way these global operators are registered.

We'll follow-up on this issue, either with a built-in debounce feature or an example in the docs. Having to put together the solution from snippets in issues is not ideal.

**Comment by HansGuckInDieLuft** [comment]
Hello Tsvetomir,

thank you; now it works: I had to import 'rxjs/add/operator/debounceTime' as your code shows,
but I also had to include "take" in the databinding.directive.d.ts, where it was missing.
(I updated the components, then updated typings, but still "take" was missing in my d.ts.
You might want to check, whether it's included in the current version.)

Once again, I am with you: hiding things is not an option.
A customer is usually prepared for many changes and will appreciate progress, but non-communicated problems will not be accepted that well.

On the contrary, from side of the  Binding Directive script, it appears to be straightforward to implement a numeric property DebounceTimeout (or similarly)
that could be set in the derived object and does that stuff. Complemented by a remark in your standard sample, that would make a much more pleasant appearance to a customer.

Kind Regards
Detlef

Detlef Hüttenbach
Senior Developer
Information Services

Phone: +49 2273 597 8472 IP-Phone: 78472, Fax: +49 2273 597 222
Email: detlef.huettenbach@computacenter.com<mailto:detlef.huettenbach@computacenter.com>

Europaring 34-40, 50170 Kerpen, Nordrhein-Westfalen, Germany
